DeprecationWarning: `duckdb.typing` is deprecated and will be removed in a future version. Please use `duckdb.sqltypes` instead.

It seems like (#514) may have needed to add some version branching here:

https://github.com/eakmanrq/sqlframe/blob/4171ad4a5d3a95e77ab748e1e748f6114b9605f6/sqlframe/duckdb/session.py#L45

I just ran into this while trying to address (https://github.com/narwhals-dev/narwhals/issues/3188) and we have a couple of failures with [warnings as errors](https://docs.python.org/3/library/warnings.html#updating-code-for-new-versions-of-dependencies) for `sqlframe`


```py
pytest --constructors="sqlframe"
E   DeprecationWarning: `duckdb.typing` is deprecated and will be removed in a future version. Please use `duckdb.sqltypes` instead.
727 failed, 1551 passed, 731 skipped, 125 xfailed in 34.64s
```

For more context, we have a similar issue with `ibis` that will be fixed once (https://github.com/ibis-project/ibis/pull/11666) gets released

```py
duckdb: 1.4.1
ibis: 10.8.0
sqlframe: 3.43.4
```

--- a/sqlframe/duckdb/session.py
+++ b/sqlframe/duckdb/session.py
@@ -7,10 +7,7 @@ from sqlframe.base.session import _BaseSession
 from sqlframe.base.util import soundex
 from sqlframe.duckdb.catalog import DuckDBCatalog
 from sqlframe.duckdb.dataframe import DuckDBDataFrame
-from sqlframe.duckdb.readwriter import (
-    DuckDBDataFrameReader,
-    DuckDBDataFrameWriter,
-)
+from sqlframe.duckdb.readwriter import DuckDBDataFrameReader, DuckDBDataFrameWriter
 from sqlframe.duckdb.table import DuckDBTable
 from sqlframe.duckdb.udf import DuckDBUDFRegistration
 
@@ -42,6 +39,7 @@ class DuckDBSession(
     def __init__(self, conn: t.Optional[DuckDBPyConnection] = None, *args, **kwargs):
         import duckdb
         from duckdb import InvalidInputException
+
         try:  # Available from duckdb 1.4.1
             from duckdb.sqltypes import VARCHAR
         except ImportError:

